widget/view.controller.js

```diff
@@ -1,233 +1,335 @@
+        $scope.feedRules.blockHighConfidenceThreatFeeds.automaticBlockIOCEnabled.threatFeedBlockSchedule.avtivateScheduleQueryParams = angular.copy(scheduleDetails);
+      } else if (!CommonUtils.isUndefined($scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromEmail.emailIngestionSchedule.avtivateScheduleQueryParams) && $scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromEmail.emailIngestionSchedule.avtivateScheduleQueryParams.name === data.scheduleDetails.name) {
+        $scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromEmail.emailIngestionSchedule.avtivateScheduleQueryParams = angular.copy(scheduleDetails);
+      }
+    });
+
+    $scope.onEmailClientChange = function (selectedEmailClient) {
+      getEmailClientDetails(selectedEmailClient.name);
+      $scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromEmail.selectedEmailClient.forEach(function (emailClient) {
+        emailClient.enabled = (emailClient.name === selectedEmailClient.name);
+      });
+    };
+
+    function toggleFeedRules(feedRule, $event) {
+      if ($event && $event.target.className === 'switch-slider') {
+        feedRule.enabled = !feedRule.enabled;
+      }
+    }
+
+    function updatesendNotification() {
+      if (!$scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromFile.enabled && !$scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromEmail.enabled) {
+        $scope.feedRules.unstructuredFeedsSupport.sendEmailNotifications.enabled = false;
+      }
+    }
+
+    function getEmailClientDetails(name) {
+      var queryBody = {
+        "sort": [
+          { "field": "featured", "direction": "DESC" },
+          { "field": "label", "direction": "ASC" }
+        ],
+        "limit": ALL_RECORDS_SIZE,
+        "logic": "AND",
+        "filters": [
+          { "field": "type", "operator": "in", "value": ["connector"] },
+          { "field": "name", "operator": "eq", "value": name },
+          { "field": "version", "operator": "notlike", "value": "%_dev" }
+        ],
+        "page": 1,
+        "__selectFields": []
+      };
+
+      $resource(API.QUERY + 'solutionpacks?$limit=' + ALL_RECORDS_SIZE + '&$page=1').save(queryBody).$promise.then(function (response) {
+        if (response['hydra:member'] && response['hydra:member'].length > 0) {
+          $scope.exchangeUUID = response['hydra:member'][0].uuid;
+        }
+      });
+    }
+
+    function _createSchedule(scheduleName, scheduleDetails, crontab) {
+      var queryBody = {
+        name: scheduleDetails.name,
+        "crontab": crontab,
+        "kwargs": {
+          "exit_if_running": true,
+          "wf_iri": API.API_3_BASE + API.WORKFLOWS + scheduleDetails.playbook_uuid,
+          "timezone": "UTC",
+          "utcOffset": "UTC",
```

Comment:
this is a reusable code, this can be moved to service code.

Reply:
USed in chain promise so can't move to service code